### PR TITLE
Adding UTIL API classes, updating tests to use them, adding test cove…

### LINKED ISSUE #594: Create global API classes for the UTIL classes in HEDA
By creating global classes, this functionality doesn't need to be recreated in orgs with HEDA installed.

### PR DESCRIPTION
**Pull request by lparrott**
…ragage for UTIL_Describe

# Critical Changes

# Changes

- New global API classes for UTIL_Describe and UTIL_UnitTestData_TEST are now available.
- All orgs with HEDA installed (including the upcoming HEDA Gift Processing) can use the utility functions in each of these classes.

# Issues Closed

- Fixes #594

# New Metadata

UTIL_Describe_API.cls
UTIL_UnitTestData_API.cls

# Deleted Metadata

# Testing Notes

- With a HEDA package (that includes this code branch) installed, ensure that methods within the new global classes (UTIL_Describe_API and UTIL_UnitTestData_API) can be called.
- Apex unit tests should cover all relevant cases, but to test manually, read on:
- Open an execute anonymous apex window from the Developer console and run UTIL_Describe_API.getObjectLabel('Account'), which should return and string and not fail
- To test UTIL_UnitTestData_API, make sure that UTIL_UnitTestData_API.isSandbox() returns 'false' in a scratch org, or 'true' from an actual Sandbox instance

### CODE REVIEW

**Comment by lparrott** [comment]
I created a package with these changes to test them as managed code, and it worked great. I can share the package with testers as needed.

File: src/classes/UTIL_Namespace_API.cls [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /*******************************************************************************************************
+    * @description Static method adds the current namespace to a string if we're in a managed context,
+    * with double underscore notation. If we're not in a managed context, returns the string itself.
+    * @param str Token name.
+    * @return Token name, with namespace prefix and double underscore if in a managed package.
+    */
+    global static String StrTokenNSPrefix(String str) {
+        return UTIL_Namespace.StrTokenNSPrefix(str);
```
Comment (weizsforg) [inline src/classes/UTIL_Namespace_API.cls]: The following method will have a dependency with getNamespace() method. Might need to update them.

File: src/classes/UTIL_Namespace_API.cls [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    
+    /*******************************************************************************************************
+    * @description Static method adds the current namespace to a string if we're in a managed context,
+    * with dot notation. If we're not in a managed context, returns the string itself.
+    * @param str Token name.
+    * @return Token name, with namespace prefix and dot notation if in a managed package.
+    */
+    global static String StrTokenNSPrefixDotNotation(String str) {
+        return UTIL_Namespace.StrTokenNSPrefixDotNotation(str);
```
Comment (weizsforg) [inline src/classes/UTIL_Namespace_API.cls]: same as above

File: src/classes/UTIL_Namespace_API.cls [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    /*******************************************************************************************************
+    * @description If we are in a managed package, tokens in dynamic SOQL must include the package namespace
+    * prefix, bit field describes don't include namespace prefix fields within the package. This method can
+    * be used in our describe code to remove an incorrect namespace prefix.
+    * Removes package prefix, then removes dynamic prefix if one is detected.
+    * @param str token name
+    * @return token name, with namespace prefix removed, if present.
+    */
+    global static String StrTokenRemoveNSPrefix(String str) {
+        return UTIL_Namespace.StrTokenRemoveNSPrefix(str);
```
Comment (weizsforg) [inline src/classes/UTIL_Namespace_API.cls]: same as above

File: src/classes/UTIL_Namespace_API.cls [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    /*******************************************************************************************************
+    * @description Static method that takes a string of a multiple potential field names or object names.
+    * If it is a custom object or field (name ends with __c), it prepends the namespace prefix if required.
+    * If we are in a managed package, tokens in dynamic SOQL must include the package namespace prefix.
+    * If you ever deploy this package as unmanaged, this routine will do nothing!
+    * @param str string that contains 0 or more token names
+    * @return same string with token names, namespace prefixed, if required.
+    */
+    global static String StrAllNSPrefix(String str) {
+        return UTIL_Namespace.StrAllNSPrefix(str);
```
Comment (weizsforg) [inline src/classes/UTIL_Namespace_API.cls]: same as above

**Comment by lparrott** [comment]
@weizsforg Thanks for the feedback, and good catch. I removed getNamespace and anything that referenced it from the UTIL_Namespace_API class, we can add those methods to the Advancement package when needed.

**Comment by weizsforg** [comment]
@lparrott Looks great!

**Comment by annejackson** [comment]
This looks good. Merging PR @lparrott!

**Pull request merged**